error.go

```diff
@@ -245,6 +249,13 @@ func NoAuthMiddleware(schemeName string) error {
 	return ErrNoAuthMiddleware(msg, "scheme", schemeName)
 }
 
+// MethodNotAllowedError is the error produced to requests that have a registered handler
+// but the used HTTP method does not match.
+func MethodNotAllowedError(method string, allowed []string) error {
+	msg := fmt.Sprintf("Method %s must be one of %s", method, strings.Join(allowed, ", "))
```

Comment:
maybe make a special case for when there's only one method? otherwise the message might be confusing (`Method GET must be one of POST`).